### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by jstucke**

### CODE REVIEW

File: .github/workflows/run_cargo_test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+          - beta
+          - nightly
+    steps:
+      - uses: actions/checkout@v4
+      - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
```
Comment (vobst) [inline .github/workflows/run_cargo_test.yml:20]: Not sure if it is worth testing on beta and nightly toolchains. Personally I'd not test on nightly, beta is ok as it gives us some headstart for changes that are likely to land.

**Review by vobst — CHANGES_REQUESTED** [review]
Looks good to me. Just see the comment about nightly.

File: .github/workflows/run_cargo_test.yml (hunk 1 above)
Comment (jstucke) [reply .github/workflows/run_cargo_test.yml:20]: I removed the nightly build

**Review by vobst — APPROVED** [review]
Thx, got nothing to nitpick anymore.

**Pull request merged**